jlm/mlir/frontend/MlirToJlmConverter.cpp

```diff
@@ -508,6 +517,58 @@ MlirToJlmConverter::ConvertOperation(
         &rvsdg::CreateOpNode<llvm::ConstantFP>(rvsdgRegion, size, constant.value()));
   }
 
+  // RVSDG does not have an index type. Indices are therefore converted to integers.
+  // The width of indices are machine dependent and 32-bits are used for x86.
+
+  else if (auto constant = ::mlir::dyn_cast<::mlir::arith::ConstantIndexOp>(&mlirOperation))
```

Comment:
A lot of constants `32` here. I would at least put them in a constant variable to make it easy to change / grep after

Reply:
I created a static GetIndexBitWidth(), which then later can be modified to provide an architecture dependent width.